### PR TITLE
Palindrome Product - factors first approach

### PR DESCRIPTION
**Pull request by michalporeba**
Creating a first approach for the [Palindrome Product](https://exercism.org/tracks/csharp/exercises/palindrome-products).
Forum: [Post your C# Approaches](https://forum.exercism.org/t/post-your-c-approaches/4758)
Related to: [an earlier palindrome product article PR](https://github.com/exercism/csharp/pull/2104)

*(It is just a draft PR to give some early visibility and, perhaps, to get an early feedback as I work on my first approach).*

### CODE REVIEW

File: exercises/practice/palindrome-products/.approaches/introduction.md [hunk 1]
```diff
@@ -1,5 +1,18 @@
 # Introduction
 
-## General guidance
+To solve the Palindrome Products exercise we first need to be able to identify if a number is a palindrome. 
+The ways in this can be done are explored in the [Is palindrome number check][is-palindrome-article] article.
+The approaches presented here assume that there is a method `bool IsPalindrome(int number)` that returns `true` if the number is a palindrome and `false` otherwise.
```
Comment (ErikSchierboom) [inline exercises/practice/palindrome-products/.approaches/introduction.md:5]: Great intro!

**Comment by michalporeba** [comment]
Hi @ErikSchierboom, I think this is the first approach ready. Could you review it? I'd like to see it live before writing another one, just to help me understand how it all looks after publishing. It is my first approach ever.

**Review by ErikSchierboom — APPROVED** [review]

File: exercises/practice/palindrome-products/.approaches/factors-first/snippet.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+```csharp
+IEnumerable<(int, int)> FactorPairsInRange(int min, int max) 
+{
+  for (var a = min; a <= max; ++a)
+    for (var b = a; b <= max; ++b)
+      yield return (a, b);
+}
+```
```
Comment (ErikSchierboom) [inline exercises/practice/palindrome-products/.approaches/factors-first/snippet.txt:8]: The snippet is rendered as-is

```suggestion
IEnumerable<(int, int)> FactorPairsInRange(int min, int max) 
{
  for (var a = min; a <= max; ++a)
    for (var b = a; b <= max; ++b)
      yield return (a, b);
}
```

File: exercises/practice/palindrome-products/.approaches/factors-first/content.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
+# Factors first
+
```
Comment (ErikSchierboom) [inline exercises/practice/palindrome-products/.approaches/factors-first/content.md:2]: We usually list the entirety of the solution in the beginning of the file.

**Review by ErikSchierboom — COMMENTED** [review]
Could of minor comments

File: exercises/practice/palindrome-products/.approaches/factors-first/content.md (hunk 3 above)
Comment (michalporeba) [reply exercises/practice/palindrome-products/.approaches/factors-first/content.md:2]: In that case I'll have to develop a full solution :)

**Comment by ErikSchierboom** [comment]
@michalporeba Where are we at with this PR? I think the only thing missing was the full code listing in the beginning.

**Comment by michalporeba** [comment]
Hi @ErikSchierboom. I need to develop the full solution in this style but had no time for it just yet. I was travelling for the last 10 days. 5 days on a conference, then some touristy stuff and then some office stuff. I might have some time at the airport tonight, or perhaps some time on a train back in UK tomorrow. If not it will have to wait until the weekend to finish.

**Comment by ErikSchierboom** [comment]
Oh sure! No rush at all.

**Comment by michalporeba** [comment]
@ErikSchierboom finally managed to get to it. The code is as long as the article itself! 
Let's see how it looks, but with so much code I think having an explanation that builds piece by piece until the code is revealed at the end could work better. But first, let's see how the current standard works.

**Review by ErikSchierboom — APPROVED** [review]
Some minor formatting things (usage of tabs instead of spaces), but let's get this merged!

**Pull request merged**